src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -108,4 +111,37 @@ protected boolean hasValidCredentials(final FilteredLog logger) {
     private Optional<GitHubAppCredentials> findGitHubAppCredentials(final String credentialsId) {
         return getScmFacade().findGitHubAppCredentials(getJob(), credentialsId);
     }
+
+    Optional<Long> getId(final String name) {
+        return getAction(name).map(GitHubChecksAction::getId);
+    }
+
+    // New API method
+    ChecksConclusion getConclusion(final String name) {
+        return getAction(name).map(GitHubChecksAction::getConclusion).orElse(ChecksConclusion.NONE);
+    }
+
+    // Implementation using invisible actions
+    private Optional<GitHubChecksAction> getAction(final String name) {
+        return job.getActions(GitHubChecksAction.class)
+                .stream()
+                .filter(a -> a.getName().equals(name))
+                .findFirst();
+    }
+
+    // Implementation fetching info on request
+    private Optional<GitHubChecksAction> getAction(final GitHub gitHub, final String name) throws IOException {
+        return gitHub.getRepository(getRepository())
+                .getCheckRuns(getHeadSha())
+                .toList()
+                .stream()
+                .filter(r -> r.getName() == name)
+                .findFirst()
+                .map(r -> new GitHubChecksAction(r.getId(), r.getName(), ChecksConclusion.NONE)); // obviously replace with logic to get ChecksConclusion from string
+
+    }
+
+    void updateAction(final long id, final String name, final ChecksConclusion conclusion) {
+        job.addOrReplaceAction(new GitHubChecksAction(id, name, conclusion));
```

Comment:
Ah, I thought you could have multiple instances of the same type, but they would be replaced based on their equality?

We could just have a single one with a mapping.

Reply:
> but they would be replaced based on their equality

it will ["Add an action, replacing any existing actions of the (exact) same class."](https://javadoc.jenkins-ci.org/hudson/model/Actionable.html#addOrReplaceAction-hudson.model.Action-)

> We could just have a single one with a mapping.

That's just a matter of design (multiple actions or one action with a mapping), but I saw some other Jenkins plugins just add actions when they need.

Reply:
Oh yes, thanks. So we would have to implement our own `addOrUpdate` effectively...are these actions mutable?